### ISSUE TITLE
As a user, I can work with large numbers of resources in the playground using pagination

### ISSUE THREAD

**Issue by fenglianxu** [hyperledger/composer#1015]
As a user, I can work with large numbers of resources in the playground using pagination, so that I can page through long lists of assets, participants, and transactions.

## Context

Previous discussion (on further down the line concepts) available here: https://github.com/hyperledger/composer/issues/396

**Comment by winslet** [comment]
@fenglianxu have you any ideas at this stage at what point you will be looking to implement this part of the query work? Trying to get an idea of when designs will be needed by.

**Comment by winslet** [comment]
We have a few ways of doing pagination:
1. Numbers on a page eg:
![image](https://cloud.githubusercontent.com/assets/13980028/26196891/4ac630c8-3bb8-11e7-8a72-8f8db07123bb.png)
2. Arrows left/right
![image](https://cloud.githubusercontent.com/assets/13980028/26196854/29828a42-3bb8-11e7-823d-5486daf9b53d.png)
3. Infinite scroll - pages load as the user reaches the bottom of the results.

**Comment by winslet** [comment]
@EdMoffatt any views on the above?

**Comment by Tobias-Hunter** [comment]
My initial thoughts are the designs need to represent how complex we would like to go with it this... 
We should display:
- Current location, what the user is currently seeing
- How many there are overall
- Ability to navigate to other locations (pages) e.g. Next & Previous

We could allow the user to:
- Select how many they wish to view
- Allow the user to select a specific location rather than Next or Previous e.g. 1st page or 6th page

My suggested happy medium: 
![image](https://cloud.githubusercontent.com/assets/25635381/26347662/2f0f35ee-3fa2-11e7-97a6-1a88bf0a8820.png)
to this:
![image](https://cloud.githubusercontent.com/assets/25635381/26347697/46527ca2-3fa2-11e7-887d-d89d03a66681.png)

🎈

**Comment by Tobias-Hunter** [comment]
How Query, Historian & Pagination will work together in the Playground, as shared on the Community call on the 1st June! 😛 

.XD Clickthrough: https://xd.adobe.com/view/b653fcaa-4e9a-40ad-8edd-96891c2210bb/

**Comment by winslet** [comment]
Hifi's for the table below, this includes the pagination (see top right) at this point we have chosen to only view 50 results in a page. There is also further work regarding the "view data" modal which shows the JSON of the transactions to come. 

### Base Table Design

![transaction registry](https://user-images.githubusercontent.com/13980028/27026115-a00039c6-4f54-11e7-81bf-5f5812206fbe.png)

### Transaction Registry - Scroll State
![transaction registry](https://user-images.githubusercontent.com/13980028/27026130-ae5879a2-4f54-11e7-9157-8fbc3044bfd5.png)

### Transaction Registry - Load more button
![transaction registry - load more](https://user-images.githubusercontent.com/13980028/27026157-c13785e0-4f54-11e7-8b03-9b3cf87655b8.png)

### Transaction Registry - Loading state

![transaction registry - loading state](https://user-images.githubusercontent.com/13980028/27026190-ce89451c-4f54-11e7-8257-0d67c19ad374.png)

*Marketch:* 
[Transaction_Registry_SW_1.zip](https://github.com/hyperledger/composer/files/1067553/Transaction_Registry_SW_1.zip)

Any questions please let me know 🎉

**Comment by fenglianxu** [comment]
Although  LIMIT/SKIP has been blocked by Fabric, playground can start to work on this as Composer runtime supports LIMIT/SKIP.  The test can not be done until Fabric supports LIMIT/SKIP.

**Comment by winslet** [comment]
@caroline-church know if there are any updates on this?

**Comment by caroline-church** [comment]
yes move to the backlog

**Comment by sstone1** [comment]
As far as I know, nobody is working on this at the moment or planning to work on it, so closing. 
If somebody wishes to work on this in the future, then please reopen this issue or open a new issue.
